edweiss-app/components/lectures/remotecontrol/modal.tsx

```diff
@@ -424,4 +425,97 @@ export const TimerSettingModal: ReactComponent<{
+
+    async function handleQuestionBroadcast() {
+        onClose();
+        if (broadcasted === id) {
+
+            // Question has been answered 
+            try {
+                await callFunction(LectureDisplay.Functions.markQuestionAsAnswered, {
+                    courseId: courseId,
+                    lectureId: lectureId,
+                    id: id,
+                    answered: true,
+                });
+                await callFunction(LectureDisplay.Functions.clearQuestionEvent, {
+                    courseId: courseId,
+                    lectureId: lectureId,
+                    id: id,
+                });
+            } catch (error) { console.error("Error updating the question event:", error); }
+
+            setBroadcasted("");
+
+        } else {
+
+            // Broadcast the question to the audience
+            try {
+                await callFunction(LectureDisplay.Functions.broadcastQuestion, {
+                    courseId: courseId,
+                    lectureId: lectureId,
+                    id: id,
+                });
+            } catch (error) { console.error("Error updating the question event:", error); }
+
+            setBroadcasted(id);
+        };
+    }
+
+    return (
+        <ModalContainer modalRef={modalRef}>
+            <>
+                <TView justifyContent='center' alignItems='center' mb='sm'>
+                    <TText bold size='lg' mb='sm'>{t('showtime:question_broadcast_modal_title')}</TText>
+                </TView>
+
+
+                <TText ml={'md'} color='overlay2' bold>{username === "" ? t('showtime:anony_ask_question') : username} {t('showtime:question_broadcast_modal_says')}</TText>
+
+
+                <TView justifyContent='center' alignItems='center' m={'md'}>
+                    <TText size={'lg'} color='overlay2' align='center'>« {question} »</TText>
+                </TView>
+
+
+                <TView flexDirection='column' alignItems='flex-end' mt='md'>
+                    {likes > 0 && (
+                        <>
+                            <TText ml={'md'} color='overlay2' mr='lg'>{likes} {t('showtime:other_student')}</TText>
+                            <TText ml={'md'} color='overlay2' mr='lg'>{t('showtime:are_interrested')}</TText>
+                        </>
+                    )}
+                </TView>
+
+                <FancyButton m='md' mb='sm' icon={broadcasted === id ? 'cloud-done-outline' : 'paper-plane-outline'} onPress={handleQuestionBroadcast} testID='brod-quest-ans-button'>
+                    {broadcasted === id ? t('showtime:question_answered') : t('showtime:broadcast_question')}
+                </FancyButton>
+
+                <FancyButton backgroundColor='subtext0' m='md' mt='sm' onPress={onClose} outlined testID='brod-quest-close-button'>
+                    {t('showtime:close_btn')}
+                </FancyButton>
```

Comment:
YOu didn't disable button when value is udated no ?

Reply:
Oh I did not knew we could do this. The reason for this is simple in this scenario: 
The `onPress` lambda will automatically call the `onClose ` which makes the modal disapear right away, this way it's impossible for the user to reclick on it as the button is gone.
Do you think that's enough as a protection ?

Reply:
Yes it's a perfect reason